tools/primreq.ml

```diff
@@ -40,16 +40,15 @@ let scan_info cu =
 
 let scan_obj filename =
   let ic = open_in_bin filename in
-  let buffer = String.create (String.length cmo_magic_number) in
+  let buffer = Bytes.create (String.length cmo_magic_number) in
```

Comment:
You should replace this and the following `really_input` by a single call to `really_input_string`, which allows to avoid the unsafe coercions below. If you do that and rebase the changes to this file in a single commit at the bottom of your commit stack (close to master), I'll push it to master now.